packages/ui-app/src/InputNumber.tsx

```diff
@@ -0,0 +1,188 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { BareProps, I18nProps } from './types';
+
+import BN from 'bn.js';
+import React from 'react';
+import isString from '@polkadot/util/is/string';
+
+import classes from './util/classes';
+import Input, { KEYS, KEYS_PRE, isCopy, isCut, isPaste, isSelectAll } from './Input';
+import translate from './translate';
+
+type Props = BareProps & I18nProps & {
+  defaultValue?: string,
+  isError?: boolean,
+  label?: any,
+  maxLength?: number,
+  onChange: (value: BN) => void,
+  placeholder?: string,
+  withLabel?: boolean
+};
+
+type State = {
+  isPreKeyDown: boolean,
+  isValid: boolean,
+  previousValue: string
+};
+
+// chain specification bit length
+const BIT_LENGTH_128 = 128;
+
+const KEYS_ALLOWED: Array<any> = [KEYS.ARROW_LEFT, KEYS.ARROW_RIGHT, KEYS.BACKSPACE, KEYS.ENTER, KEYS.ESCAPE, KEYS.TAB];
+
+class InputNumber extends React.PureComponent<Props, State> {
+  state: State = {
+    isPreKeyDown: false,
+    isValid: false,
+    previousValue: '0'
+  };
+
+  render () {
+    const { className, defaultValue, maxLength, style, t } = this.props;
+    const { isValid, previousValue } = this.state;
+    const revertedValue = !isValid ? previousValue : undefined;
+
+    return (
+      <div
+        className={classes('ui--InputNumber', className)}
+        style={style}
+      >
+        <Input
+          {...this.props}
+          defaultValue={defaultValue || '0'}
+          maxLength={maxLength || this.maxLength()}
```

Comment:
Here maxLength is fine, however we use the bitLength as passed in to calculate to pass to Input. So InpoutNumber works with bitLength, the lower-level can work with character lengths.